#include <iostream>
#include <string>
#include <stdio.h>
using namespace std;

void basic_handling(){
	try{
		cout << "Throwing an exception now: ";
		string msg = "Error message";
		throw msg;
	}

	catch(char const* ExceptionMessage){
		printf("%s\n", ExceptionMessage);
	}

	catch(string ExceptionMessage){
		cout << ExceptionMessage << endl;
	}

	catch(...){
		cerr << "Generic catch statement" << endl;
	}

	cout << "Things here are equivalent to a finally block\n";
}

int main(){
	basic_handling();
}
